### PR TITLE
Assemble & Integrate Extractors

### PR DESCRIPTION
**Pull request by wcjordan**
**What it accomplishes:**

* Wire prior steps into a single `extract_features(chunk)` function that returns a populated `FeatureChunk`.
* Preserve existing `Chunk` inputs and do not alter preprocessing outputs.

**How to verify:**

* End-to-end tests against sample chunks confirm that each feature list (`dom_mutations`, `interactions`, `delays`, etc.) is non-empty and matches expected counts/values.
* Regression tests ensure feature extraction does not throw errors or mutate input chunks.

--
Implements the end-to-end feature extraction pipeline by wiring together individual extractors into a single extract_features function, adds comprehensive integration tests, and updates the documentation checklist to reflect completion.

* Introduce extract_features in pipeline.py that applies mutations, extracts features, and assembles them into a FeatureChunk.
* Add integration tests in test_pipeline_features.py to validate each feature category and ensure idempotency and proper handling of edge cases.
* Mark pipeline-related tasks as done in todo.md.

### CODE REVIEW

File: test_gen/feature_extraction/tests/test_pipeline_features.py [hunk 1]
```diff
@@ -0,0 +1,425 @@
...
+from feature_extraction.models import UINode
+
+
+@pytest.fixture(name="sample_chunk_with_interactions")
+def fixture_sample_chunk_with_interactions():
+    """Fixture providing a sample Chunk with various interaction and mutation events."""
+    events = [
+        # FullSnapshot event (for DOM initialization)
+        {
+            "type": 2,
```
[the remark this answers is not in the document]
Comment (wcjordan) [reply test_gen/feature_extraction/tests/test_pipeline_features.py:22]: Added a TODO for this
